tests/explore_test.sh

```diff
@@ -5,21 +5,111 @@
 
 function suite
 {
-  suite_addTest "testExplore"
+  suite_addTest "explore_files_under_git_repo_Test"
+  suite_addTest "explore_git_log_Test"
+  suite_addTest "explore_parser_Test"
 }
 
-function testExplore
+function setupExplore()
 {
-  local MSG_OUT="GNU"
-  local LOG_OUT="Initial commit"
+  local -r test_path="tests/.tmp"
+  local -r current_path="$PWD"
 
-  assertEquals "Expected an error message." "Expected path or 'log'" "$(explore)"
+  rm -rf "$test_path"
+
+  mkdir -p "$test_path"
+  cd "$test_path"
+  # Setup git repository for test
+  git init &> /dev/null
+
+  for commit in {1..5}; do
+    local file_name="file_$commit.txt"
+    echo "This is file $commit" > "$file_name"
+    git add "$file_name" &> /dev/null
+    git commit -m "Commit number $commit" &> /dev/null
+  done
+
+  cd "$current_path"
+}
+
+function tearDownExplore()
+{
+  local -r test_path="tests/.tmp"
+
+  rm -rf "$test_path"
+}
+
+function explore_files_under_git_repo_Test
+{
+  local ID
+  local MSG_OUT
+
+  setupExplore
+
+  ID=1
+  MSG_OUT="Expected string or 'log'"
+  output=$(explore)
+  assertEquals "($ID) - Expected an error message." "$MSG_OUT" "$output"
+
+  ID=2
   MSG_OUT="camel_case(void)"
-  assertEquals "$MSG_OUT" "$(explore "camel_case" tests/samples | cut -d ' ' -f2 | sed -n -e 2p)"
-  assertEquals "$LOG_OUT" "$(explore log LICENSE | grep "Initial commit" | awk '{print $1, $2}')"
+  output=$(explore "camel_case" tests/samples | cut -d ' ' -f2 | sed -n -e 2p)
+  assertEquals "($ID)" "$MSG_OUT" "$output"
+
+  ID=3
   MSG_OUT="Free Software Foundation"
-  assertEquals "$MSG_OUT" "$(explore "$MSG_OUT" ./LICENSE | grep "$MSG_OUT" -o | head -n 1)"
-  true
+  output=$(explore "$MSG_OUT" ./LICENSE | grep "$MSG_OUT" -o | head -n 1)
+  assertEquals "($ID)" "$MSG_OUT" "$output"
+
+  tearDownExplore
+}
+
+function explore_git_log_Test
+{
+  local ID
+  local file_name
+  local commit_msg
+  local -r test_path="tests/.tmp"
+  local -r current_path="$PWD"
+
+  setupExplore
+  cd "$test_path"
+
+  ID=1
+  commit_msg="Commit number 2"
+  file_name="file_2.txt"
+  output=$(explore --log "$commit_msg" | grep "$commit_msg" | awk '{print $1, $2, $3}')
+  assertEquals "($ID)" "$commit_msg" "$output"
+
+  cd "$current_path"
+  tearDownExplore
+}
+
+function explore_parser_Test
+{
+  local ID
+
+  # Expected behaviour
+  ID=1
+  output=$(explore_parser --log)
+  ret="$?"
+  assertEquals "($ID)" "1" "$ret"
+
+  ID=2
+  output=$(explore_parser "something")
+  ret="$?"
+  assertEquals "($ID)" "2" "$ret"
+
+  # Others
+  ID=3
+  output=$(explore_parser --logljkl)
+  ret="$?"
+  assertEquals "($ID)" "2" "$ret"
```

Comment:
I think we should consider all `--[A-Za-z]` as invalid options. Then, if the user really wants to use it as a search regex, they can use `\--[A-Za-z]`. The advantage is that it will be then possible to search for the string "\--log", for example, without confusing it with the option.

Reply:
Any thought on this idea, @rodrigosiqueira ?

Reply:
Try `kw e --log "--log"`in the linux kernel, it works :)
Actually, I'm just testing the correct use of the flag `--log` in this test.